<launch>
	<param name="use_sim_time" value="true"/>

	<node name="gazebo" pkg="gazebo_ros" type="gzserver" args="$(find hector_nist_arena_worlds)/worlds/exp3.world" /> 

  	<!-- <node name="gazebo_gui" pkg="gazebo_ros" type="gzclient"/> -->

	<include file="$(find p3at_description)/launch/spawn_p3at.launch">
		<arg name="robot_name" value="rtg"/>
		<arg name="pose_args" value="-x -1.2 -y 0 -Y -1.57"/> 
	</include>

	<node name="keyboard_drive" pkg="proje" type="keyboard_cmdvel.py" output = "screen"/>

	<node name="gezinge" pkg="proje" type="trajectory"/>

	<node pkg="tf" type="static_transform_publisher" name="world2trajectory" args="0 0 0 0 0 0 /map /trajectory 100"/>

	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find proje)/rviz/proje.rviz"/>

	<!-- BU LAUNCH DOSYASINA BURDAN SONRA EKLEME YAPABILIRSINIZ -->
	<!-- <node name= "camera_read" pkg= "proje" type="camera_read.py" output="screen" required = "true"/> -->
	<node name= "camera_read" pkg= "proje" type="camera_read.py" output="screen" required = "true">
		<param name="hazmat_dir" value="$(find proje)/img/hazmat_template/"/>
		<param name="wechat_dir" value="$(find proje)/model/wechat-qrcode/"/>
	</node>

	<!-- MAPPING -->
	<!-- <include file="$(find proje)/launch/mapping.launch">
		<arg name="scan_topic"  value="/rtg/hokuyo" />
		<arg name="base_frame"  value="/rtg/base_link"/>
		<arg name="odom_frame"  value="/rtg/odom"/>
	</include> -->
	<arg name="scan_topic"  default="/rtg/hokuyo" />
	<arg name="base_frame"  default="/rtg/base_link"/>
	<arg name="odom_frame"  default="/rtg/odom"/>

	<node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
		<param name="use_sim_time" value="true"/>
		<param name="base_frame" value="$(arg base_frame)"/>
		<param name="odom_frame" value="$(arg odom_frame)"/>
		<remap from="scan" to="$(arg scan_topic)"/>
		<param name="map_update_interval" value="3.0"/>
		<param name="maxUrange" value="6.0"/>
		<param name="maxRange" value="8.0"/>
		<param name="sigma" value="0.05"/>
		<param name="kernelSize" value="1"/>
		<param name="lstep" value="0.05"/>
		<param name="astep" value="0.05"/>
		<param name="iterations" value="5"/>
		<param name="lsigma" value="0.075"/>
		<param name="ogain" value="3.0"/>
		<param name="lskip" value="0"/>
		<param name="minimumScore" value="200"/>
		<param name="srr" value="0.01"/>
		<param name="srt" value="0.02"/>
		<param name="str" value="0.01"/>
		<param name="stt" value="0.02"/>
		<param name="linearUpdate" value="0.5"/>
		<param name="angularUpdate" value="0.436"/>
		<param name="temporalUpdate" value="-1.0"/>
		<param name="resampleThreshold" value="0.5"/>
		<param name="particles" value="80"/>
		<param name="xmin" value="-10.0"/>
		<param name="ymin" value="-10.0"/>
		<param name="xmax" value="10.0"/>
		<param name="ymax" value="10.0"/>
		<param name="delta" value="0.05"/>
		<param name="llsamplerange" value="0.01"/>
		<param name="llsamplestep" value="0.01"/>
		<param name="lasamplerange" value="0.005"/>
		<param name="lasamplestep" value="0.005"/>
	</node>

	<!-- Object Marker -->
	<node name="mark_object" pkg="proje" type="mark_object.py" output="screen"/>


	<!-- BURAYA KADAR -->	
</launch>
